### PR TITLE
kubelet/rkt: Add getImageManifest()

### PR DESCRIPTION
**Pull request by yifan-gu**
Assume that https://github.com/coreos/rkt/pull/901 got merged, then we can run a pod using rkt for real!

I will test out other pods in the example dir. 
Also as discussed with @dchen1107 offline, think we should configure the network in case the pods are running in private network as well.

/cc @dchen1107 @vmarmol @yujuhong @jonboulle @bakins

### CODE REVIEW

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -303,7 +304,7 @@ func setApp(app *appctypes.App, c *api.Container) error {
 
 	// Override the environment.
 	// TODO(yifan): Use RunContainerOptions.
```
Comment (bakins) [inline pkg/kubelet/rkt/rkt.go]: can remove this TODO?

File: pkg/kubelet/rkt/rkt.go [hunk 2]
```diff
@@ -349,9 +351,26 @@ func setApp(app *appctypes.App, c *api.Container) error {
...
+// for the image.
+func (r *runtime) getImageManifest(image string) (*appcschema.ImageManifest, error) {
+	var manifest appcschema.ImageManifest
+
+	// TODO(yifan): Assume docker images for now.
+	output, err := r.runCommand("image", "cat-manifest", "--quiet", dockerPrefix+image)
+	if err != nil {
+		return nil, err
+	}
+	if len(output) != 1 {
```
Comment (vmarmol) [inline pkg/kubelet/rkt/rkt.go]: Why is the output one character?

File: pkg/kubelet/rkt/rkt.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: @bakins ack

File: pkg/kubelet/rkt/rkt.go (hunk 2 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: @vmarmol That's one line actually, I will update the comments of the runCommand()

Comment (vmarmol) [inline pkg/kubelet/rkt/rkt.go]: Aaah right it outputs lines, thank you :)

**Comment by vmarmol** [comment]
LGTM, will merge on green.

**Comment by vmarmol** [comment]
I hope Shippable tries to pick it up. Waiting for Travis will take forever :(

File: pkg/kubelet/rkt/rkt.go [hunk 3]
```diff
@@ -174,7 +175,7 @@ func (r *runtime) buildCommand(args ...string) *exec.Cmd {
 }
 
 // runCommand invokes rkt binary with arguments and returns the result
-// from stdout in a list of strings.
+// from stdout in a list of strings. Each element in the string is a line.
```
Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: "Each string in the list is a line" or "Each element in the list is a line"?

File: pkg/kubelet/rkt/rkt.go [hunk 4]
```diff
@@ -349,9 +350,26 @@ func setApp(app *appctypes.App, c *api.Container) error {
...
+func (r *runtime) getImageManifest(image string) (*appcschema.ImageManifest, error) {
+	var manifest appcschema.ImageManifest
+
+	// TODO(yifan): Assume docker images for now.
+	output, err := r.runCommand("image", "cat-manifest", "--quiet", dockerPrefix+image)
+	if err != nil {
+		return nil, err
+	}
+	if len(output) != 1 {
+		return nil, fmt.Errorf("no output")
```
Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: Is it possible to have more than one line of output? I am just curious...

Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: (It might be helpful to include the output in the error message)

Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: It shouldn't, I redirect the stdout to null in rkt in that PR. But I guess "invalid output + the real output" sounds better. Thank you!

File: pkg/kubelet/rkt/rkt.go (hunk 3 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: Good catch. Thank you!

**Pull request merged**